✨ Release 1.0.0 ✨ (#6877)
ZcashFoundation/zebra: CHANGELOG.md

> and this project adheres to [Semantic Versioning](https://semver.org).

Reviewer: The release checklist in the PR description is out of date, did you want to update it, or open a new ticket with the latest steps?
https://github.com/ZcashFoundation/zebra/blob/main/.github/PULL_REQUEST_TEMPLATE/release-checklist.md#release-zebra

Author: Just rebased on main

Reviewer: Rebasing doesn't change the release checklist in this PR's description, which we are following to tag & test this release.
<img width="893" alt="Screenshot 2023-06-14 at 08 33 16" src="https://github.com/ZcashFoundation/zebra/assets/8951843/5693d340-055f-466f-bec7-a0469b26126e">

Would it help if I updated this PR's description?

## [Zebra 1.0.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0) - 2023-06-14

> This is our 1.0.0 stable release.

Author: ```suggestion
This is our 1.0.0 stable release.

This release also fixes a panic at startup when parsing the app version, begins publishing to crates.io, and publishes to Docker Hub under the `latest` tag.
```

and this project adheres to [Semantic Versioning](https://semver.org).

> ## [Zebra 1.0.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0) - 2023-06-14

Reviewer: Should these changes be in the changelog?
- #6908
- #6892
- #6890 (this actually changes the production Docker image)

Is it worth adding a comment to the PR with a list of deleted changelog entries? (I don't mind either way.)

Author: Maybe 6892, not sure about 6890

Author: > Should these changes be in the changelog?
> 
> * [change(crates): Prepare for publishing crates to crates.io #6908](https://github.com/ZcashFoundation/zebra/pull/6908)
> * [fix(rust): Declare the correct compatibility with older Rust versions #6892](https://github.com/ZcashFoundation/zebra/pull/6892)
> * [fix(deployment): Fix log file path and log colour #6890](https://github.com/ZcashFoundation/zebra/pull/6890) (this actually changes the production Docker image)
> 
> Is it worth adding a comment to the PR with a list of deleted changelog entries? (I don't mind either way.)

Release Drafter draft:

```

## [Zebra 1.0.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0) - *TODO*: date

This release *TODO*: a summary of the significant user-visible changes in the release

### Breaking Changes

This release has the following breaking changes:
- *TODO*: Check the `Removed` section for any breaking changes
- *TODO*: Add a short description of the user impact of each breaking change, and any actions users need to take
  
- build(deps): update locked dependencies ([#6878](https://github.com/ZcashFoundation/zebra/pull/6878))

### Security

- fix(deploy): Stop doing another full sync on every main branch push ([#6895](https://github.com/ZcashFoundation/zebra/pull/6895))
- fix(net): Avoid potential concurrency bugs in outbound handshakes ([#6869](https://github.com/ZcashFoundation/zebra/pull/6869))

### Added

- add(doc): Add instructions for doing mass renames easily ([#6865](https://github.com/ZcashFoundation/zebra/pull/6865))
- docs(readme): add user support section ([#6873](https://github.com/ZcashFoundation/zebra/pull/6873))

### Changed

- add(doc): Add instructions for doing mass renames easily ([#6865](https://github.com/ZcashFoundation/zebra/pull/6865))
- build(deps): bump tj-actions/changed-files from 36.0.18 to 36.1.0 ([#6874](https://github.com/ZcashFoundation/zebra/pull/6874))
- change(cd): Deploy testnet instances for every main branch push and release ([#6842](https://github.com/ZcashFoundation/zebra/pull/6842))
- change(checkpoints): Update checkpoints for 1.0.0 ([#6884](https://github.com/ZcashFoundation/zebra/pull/6884))
- change(crates): Prepare for publishing crates to crates.io ([#6908](https://github.com/ZcashFoundation/zebra/pull/6908))
- change: Rename tower-batch to tower-batch-control ([#6907](https://github.com/ZcashFoundation/zebra/pull/6907))

### Fixed

- ci(tests): run network syncs on a weekly schedule ([#6920](https://github.com/ZcashFoundation/zebra/pull/6920))
- fix(app): Stop panicking at startup when parsing the app version - release blocker ([#6888](https://github.com/ZcashFoundation/zebra/pull/6888))
- fix(ci): Disable scheduled testnet full sync by default ([#6930](https://github.com/ZcashFoundation/zebra/pull/6930))
- fix(deploy): Stop doing another full sync on every main branch push ([#6895](https://github.com/ZcashFoundation/zebra/pull/6895))
- fix(deploy): allow the container to raise in MIGs ([#6893](https://github.com/ZcashFoundation/zebra/pull/6893))
- fix(deployment): Fix log file path and log colour ([#6890](https://github.com/ZcashFoundation/zebra/pull/6890))
- fix(net): Avoid potential concurrency bugs in outbound handshakes ([#6869](https://github.com/ZcashFoundation/zebra/pull/6869))
- fix(release): Fix incorrect commands in release-checklist.md ([#6883](https://github.com/ZcashFoundation/zebra/pull/6883))
- fix(rust): Declare the correct compatibility with older Rust versions ([#6892](https://github.com/ZcashFoundation/zebra/pull/6892))
- fix(test): Avoid a race condition in testing modified JoinSplits ([#6921](https://github.com/ZcashFoundation/zebra/pull/6921))
- release(docs): Refactor bug report template ([#6858](https://github.com/ZcashFoundation/zebra/pull/6858))

### Trivial *TODO:* put this in a PR comment, not the CHANGELOG

- add(doc): Add instructions for doing mass renames easily ([#6865](https://github.com/ZcashFoundation/zebra/pull/6865))
- build(deps): Upgrade to ed25519-zebra 4.0.0 ([#6881](https://github.com/ZcashFoundation/zebra/pull/6881))
- build(deps): bump actions/checkout from 3.5.2 to 3.5.3 ([#6901](https://github.com/ZcashFoundation/zebra/pull/6901))
- build(deps): bump clap from 4.3.2 to 4.3.3 ([#6904](https://github.com/ZcashFoundation/zebra/pull/6904))
- build(deps): bump docker/build-push-action from 4.0.0 to 4.1.0 ([#6902](https://github.com/ZcashFoundation/zebra/pull/6902))
- build(deps): bump docker/login-action from 2.1.0 to 2.2.0 ([#6849](https://github.com/ZcashFoundation/zebra/pull/6849))
- build(deps): bump docker/metadata-action from 4.4.0 to 4.5.0 ([#6850](https://github.com/ZcashFoundation/zebra/pull/6850))
- build(deps): bump serde from 1.0.163 to 1.0.164 ([#6875](https://github.com/ZcashFoundation/zebra/pull/6875))
- build(deps): bump tj-actions/changed-files from 36.0.18 to 36.1.0 ([#6874](https://github.com/ZcashFoundation/zebra/pull/6874))
- change(cd): Deploy testnet instances for every main branch push and release ([#6842](https://github.com/ZcashFoundation/zebra/pull/6842))
- change(checkpoints): Update checkpoints for 1.0.0 ([#6884](https://github.com/ZcashFoundation/zebra/pull/6884))
- ci(tests): run network syncs on a weekly schedule ([#6920](https://github.com/ZcashFoundation/zebra/pull/6920))
- docs: Prep release checklist for stable ([#6876](https://github.com/ZcashFoundation/zebra/pull/6876))
- fix(ci): Disable scheduled testnet full sync by default ([#6930](https://github.com/ZcashFoundation/zebra/pull/6930))
- fix(deploy): Stop doing another full sync on every main branch push ([#6895](https://github.com/ZcashFoundation/zebra/pull/6895))
- fix(deploy): allow the container to raise in MIGs ([#6893](https://github.com/ZcashFoundation/zebra/pull/6893))
- fix(deployment): Fix log file path and log colour ([#6890](https://github.com/ZcashFoundation/zebra/pull/6890))
- fix(release): Fix incorrect commands in release-checklist.md ([#6883](https://github.com/ZcashFoundation/zebra/pull/6883))
- release(docs): Refactor bug report template ([#6858](https://github.com/ZcashFoundation/zebra/pull/6858))

### Contributors

Thank you to everyone who contributed to this release, we couldn't make Zebra without you:
@dconnolly, @dependabot[bot], @gustavovalverde, @oxarbitrage, @teor2345 and @upbqdn

```

> This is our 1.0.0 stable release.

Reviewer: > TODO: a summary of the significant user-visible changes in the release

#6888 seems like a critical fix for some users
#6908 and the Dockerhub images at the `latest` tag might be worth calling attention to

Author: #6888 is in there: https://github.com/ZcashFoundation/zebra/pull/6877/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR26

Reviewer: I'd like to make sure users know about these changes, by putting them in the release summary at the top of the changelog.

Here's an example from a previous release:
https://github.com/ZcashFoundation/zebra/blob/d2a29b655257e3ac947199098ccc85af50cca92f/CHANGELOG.md?plain=1#L9-L12

Author: I added 4b37f5a but wordsmithing welcome

Please report bugs to [the Zebra GitHub repository](https://github.com/ZcashFoundation/zebra/issues/new?assignees=&labels=C-bug%2C+S-needs-triage&projects=&template=bug_report.yml&title=)

### Security

- Avoid potential concurrency bugs in outbound handshakes ([#6869](https://github.com/ZcashFoundation/zebra/pull/6869))

### Changed

- Rename tower-batch to tower-batch-control ([#6907](https://github.com/ZcashFoundation/zebra/pull/6907))
- Upgrade to ed25519-zebra 4.0.0 ([#6881](https://github.com/ZcashFoundation/zebra/pull/6881))

### Fixed

> - Stop panicking at startup when parsing the app version - release blocker ([#6888](https://github.com/ZcashFoundation/zebra/pull/6888))

Reviewer: ```suggestion
- Stop panicking at startup when parsing the app version ([#6888](https://github.com/ZcashFoundation/zebra/pull/6888))
```

> - Allow the container to raise in MIGs ([#6893](https://github.com/ZcashFoundation/zebra/pull/6893))

Reviewer: Not user-visible
```suggestion
```

and this project adheres to [Semantic Versioning](https://semver.org).

## [Zebra 1.0.0](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0) - 2023-06-14

This is our 1.0.0 stable release.

> This release also fixes a panic at startup when parsing the app version, begins publishing to crates.io, and publishes to Docker Hub under the `latest` tag.

Reviewer: ```suggestion
This release also fixes a panic at startup when parsing the app version, [publishes `zebrad` to crates.io](https://crates.io/crates/zebrad), and [publishes to Docker Hub under the `latest` tag](https://hub.docker.com/r/zfnd/zebra/tags).
```

Please report bugs to [the Zebra GitHub repository](https://github.com/ZcashFoundation/zebra/issues/new?assignees=&labels=C-bug%2C+S-needs-triage&projects=&template=bug_report.yml&title=)

### Security

- Avoid potential concurrency bugs in outbound handshakes ([#6869](https://github.com/ZcashFoundation/zebra/pull/6869))

### Changed

> - Rename tower-batch to tower-batch-control ([#6907](https://github.com/ZcashFoundation/zebra/pull/6907))

Author: ```suggestion
- Publish to [crates.io](https://crates.io/crates/zebrad) ([#6908(https://github.com/ZcashFoundation/zebra/pull/6908))
- Rename tower-batch to tower-batch-control ([#6907](https://github.com/ZcashFoundation/zebra/pull/6907))
```

> and this project adheres to [Semantic Versioning](https://semver.org).

Reviewer: To bump the tower crates:
```sh
CARGO_UNSTABLE_SPARSE_REGISTRY=true cargo release version --verbose --package tower-batch-control --package tower-fallback --execute beta
```

Author: Done: 
<img width="826" alt="image" src="https://github.com/ZcashFoundation/zebra/assets/552961/6cf0efbd-3316-4c5e-aabc-14c2c83fbbb5">
